js/stores/windowStore.js

```diff
@@ -750,6 +752,22 @@ const doAction = (action) => {
         }
         break
       }
+    case windowConstants.WINDOW_ON_TOGGLE_BOOKMARK:
+      {
+        // show the AddEditBookmarkHanger control; saving/deleting takes place there
+        const activeFrame = frameStateUtil.getActiveFrame(windowState)
+        let siteDetail = siteUtil.getDetailFromFrame(activeFrame, siteTags.BOOKMARK)
+        const key = siteUtil.getSiteKey(siteDetail)
+
+        if (key !== null) {
```

Comment:
that's true- maybe it's OK as-is (for this PR) and that would be the place to tackle it 😄 

It should have everything (the example above) because (in the current code) whatever is found is copied from appState.  So instead of copying it, just use the entire object (which like you said, has favicon, tags, themeColor, etc).  This is great because if we needed to expose those we could.  By not using the real object, we're opening ourselves up to bugs.  For example, what if `parentFolderId` is changed to `parentDirectoryId`. If we properly handled it, we'd only need to update in `siteUtil`.  But because this code is copying the object, we'd have to update in 2 (or more) places